content/en/docs/setup/independent/troubleshooting-kubeadm.md

```diff
@@ -215,27 +215,64 @@ kubectl -n kube-system patch --type json daemonset kube-proxy -p "$(cat <<'EOF'
 [
     {
         "op": "add",
-        "path": "/spec/template/spec/containers/0/env",
-        "value": [
-            {
-                "name": "NODE_NAME",
-                "valueFrom": {
-                    "fieldRef": {
-                        "apiVersion": "v1",
-                        "fieldPath": "spec.nodeName"
-                    }
-                }
+        "path": "/spec/template/spec/volumes/-",
+        "value": {
+                "name": "shared-data",
+                "mountPath": "/shared-data"
+            }
+    },
+    {
+        "op": "add",
+        "path": "/spec/template/spec/containers/0/volumeMounts/-",
+        "value": {
+                "name": "shared-data",
+                "mountPath": "/shared-data"
             }
-        ]
+    },
+    {
+        "op": "replace",
+        "path": "/spec/template/spec/containers/0/command/1",
+        "value": "--config=/shared-data/config.conf"
     },
     {
         "op": "add",
-        "path": "/spec/template/spec/containers/0/command/-",
-        "value": "--hostname-override=${NODE_NAME}"
+        "path": "/spec/template/spec/initContainers",
+        "value": [
+                {
+                    "command": [
+                        "sh",
+                        "-c",
+                        "/bin/sed \"s/hostnameOverride: \\\"\\\"/hostnameOverride: $(NODE_NAME)/\" /var/lib/kube-proxy/config.conf > /shared-data/config.conf"
+                    ],
+                    "env": [
+                        {
+                            "name": "NODE_NAME",
+                            "valueFrom": {
+                                "fieldRef": {
+                                    "apiVersion": "v1",
+                                    "fieldPath": "spec.nodeName"
+                                }
+                            }
+                        }
+                    ],
+                    "image": "k8s.gcr.io/kube-proxy-amd64:v1.11.1",
```

Comment:
Might be good to add a note that the arch and version is hardcoded here.

Reply:
it's JSON doesn't allow comments, so it should go bellow the JSON example.

> Please replace `amd64` and `v1.11.1` in `kube-proxy-amd64:v1.11.1` with the architecture and version you wish to use for the kube-proxy image.

Reply:
We could use "busybox" instead, sine all it's doing is invoking _sed_.

Reply:
I chose to use the same image for the init container and kube-proxy container because the busybox image might not be available. I am applying this patch on an air-gapped cluster where busybox is not present.

Air-gapped clusters are not the common use case, so I think it would be ok to use busybox by default and add a note to replace busybox with the kube-proxy image if busybox is not available.

I can also think of some way to derive the correct kube-proxy image at the time the patch is applied, but that might be over-engineering.

Reply:
@detiber Do you have an opinion on whether the patch should use the `busybox` image?

Reply:
@dlipovetsky On one hand I like the idea of using a minimal image for the initContainer, but on the other I do like the idea of not requiring an additional image for air-gapped environments. I'm not sure that either approach has a clear advantage for a temporary workaround...

Reply:
"Temporary" to a point: as of now we don't know who will work on making kube-proxy's `--hostname-override` flag work again.

Regarding the air-gapped situation, I build my machine images with _busybox_ pulled, and use the "imagePullPolicy" field with value "IfNotPresent," so that the init container here won't pull the image again. We could achieve the same thing by specifying an image tag.

Reply:
@seh are you able to submit a PR with your proposed changes?

Reply:
Yes, I can do that tomorrow morning.

Reply:
Please see #9878.